### PR TITLE
Make Mutation Operators configurable

### PR DESCRIPTION
**Pull request by Quolyk**
#39
- [x] First need to merge #88
- [x] Handle validation of a config input
- [x] Propagate the valid config values of `getMutationOperators` to `GoogleTestFinder`.
- [x] Adjust mull-driver target to `GoogleTestFinder` constructor

### CODE REVIEW

**Comment by stanislaw** [comment]
@Quolyk @AlexDenisov 

I have added 1 commit for parsing mutation operators from a config YAML file:

https://github.com/mull-project/mull/tree/config_mutation

Now we need to discuss:
- how to handle validation of a config input
- how to propagate the valid config values of `getMutationOperators` to `GoogleTestFinder`.

**Comment by AlexDenisov** [comment]
My suggestions:

> how to handle validation of a config input

Print an error message if there is no such mutation operator as described in the config?

> how to propagate the valid config values of getMutationOperators to GoogleTestFinder.

Inject via a constructor?

> Is it Config class who is responsible for the conversion of mutation operators as strings to mutation operators as objects or that is Driver?

I would say it is a Driver's responsibility. But we can change it at any point if it doesn't work well.

**Comment by Quolyk** [comment]
@stanislaw my bad

**Comment by stanislaw** [comment]
@Quolyk now it looks good. I'll test this against OrcJIT today and merge if all is fine.

**Comment by stanislaw** [comment]
@Quolyk could you please also rebase into 1 commit?

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ cd ~/Projects/LLVM
 
 git clone http://llvm.org/git/llvm.git
 cd llvm
-git checkout 106bb1fe2db88fe77458ec594c461c8428588dab
+git checkout dc30ded6fb9526aba31a86b97e8f69fc9baece00
```
Comment (stanislaw) [inline README.md]: These changes should be here since you rebased from master which has these strings updated, no?

File: README.md [hunk 2]
```diff
@@ -37,5 +37,5 @@ Update to LLVM's trunk should be a recurrent task, but first we need to get
 experience of living in a downstream world.
 
 Current code is being developed against LLVM commit:
-[106bb1fe2db88fe77458ec594c461c8428588dab](https://github.com/llvm-mirror/llvm/commit/106bb1fe2db88fe77458ec594c461c8428588dab).
+[dc30ded6fb9526aba31a86b97e8f69fc9baece00](https://github.com/llvm-mirror/llvm/commit/dc30ded6fb9526aba31a86b97e8f69fc9baece00).
```
Comment (stanislaw) [inline README.md]: The same as above.

File: lib/GoogleTest/GoogleTestFinder.cpp [hunk 3]
```diff
@@ -98,12 +95,12 @@ std::vector<std::unique_ptr<Test>> GoogleTestFinder::findTests(Context &Ctx) {
         continue;
       }
 
-      Type *SeqTy = Ty->getSequentialElementType();
```
Comment (stanislaw) [inline lib/GoogleTest/GoogleTestFinder.cpp]: This also looks like a rebase artifact. The master branch should now have `getPointerElementType` (https://github.com/mull-project/mull/blob/master/lib/GoogleTest/GoogleTestFinder.cpp#L101).

File: lib/ModuleLoader.cpp [hunk 4]
```diff
@@ -3,8 +3,7 @@
 #include "Logger.h"
 
 #include "llvm/AsmParser/Parser.h"
-//#include "llvm/Bitcode/BitcodeReader.h"
-#include "llvm/Bitcode/ReaderWriter.h"
+#include "llvm/Bitcode/BitcodeReader.h"
```
Comment (stanislaw) [inline lib/ModuleLoader.cpp]: The same: rebase artefact. Master already has this change.

**Comment by Quolyk** [comment]
@stanislaw Sorry for the issues. I need to read that git book again.

**Comment by stanislaw** [comment]
Now it looks perfect :)

Since we still don't have integration tests in place (`lit`) I will have to test your branch manually against something real like `OrcJIT`. After that we'll merge.

**Comment by AlexDenisov** [comment]
> Sorry for the issues

@Quolyk you don't have to say this, we are here to help each other ;)

**Pull request merged**